File: test_resources.py
import urllib.request
from html.parser import HTMLParser
from urllib.parse import urljoin

class ResourceParser(HTMLParser):
    def __init__(self, base_url):
        super().__init__()
        self.base_url = base_url
        self.resources = []
        self.base_tag_href = None

    def handle_starttag(self, tag, attrs):
        attrs = dict(attrs)
        if tag == 'base' and 'href' in attrs:
            self.base_tag_href = attrs['href']
        
        url = None
        if tag == 'link' and 'stylesheet' in attrs.get('rel', ''):
            url = attrs.get('href')
        elif tag == 'script' and 'src' in attrs:
            url = attrs.get('src')
        elif tag == 'img' and 'src' in attrs:
            url = attrs.get('src')
            
        if url:
            self.resources.append(url)

def test_page(page_url):
    print(f"\n--- Testing {page_url} ---")
    req = urllib.request.Request(page_url, headers={'User-Agent': 'Mozilla/5.0'})
    try:
        response = urllib.request.urlopen(req)
        html = response.read().decode('utf-8')
    except Exception as e:
        print(f"Failed to fetch {page_url}: {e}")
        return

    parser = ResourceParser(page_url)
    parser.feed(html)
    
    base = parser.base_tag_href
    print(f"Base tag found: {base}")
    
    base_url = urljoin(page_url, base) if base else page_url
    print(f"Resolved base URL for relative paths: {base_url}")
    
    for res in parser.resources:
        if res.startswith('http'):
            full_url = res
        else:
            full_url = urljoin(base_url, res)
            
        try:
            req_res = urllib.request.Request(full_url, headers={'User-Agent': 'Mozilla/5.0'}, method='HEAD')
            res_response = urllib.request.urlopen(req_res)
            print(f"[OK] {res_response.status} {full_url}")
        except Exception as e:
            print(f"[ERROR] {e} -> {full_url}")

test_page('https://dev.mkavs.com/Pricing')
test_page('https://dev.mkavs.com/Branding')
test_page('https://dev.mkavs.com/Support')

File: test_resources.js
const https = require('https');
const http = require('http');

function checkUrl(url) {
    return new Promise((resolve) => {
        const req = (url.startsWith('https') ? https : http).request(url, { method: 'HEAD', headers: { 'User-Agent': 'Mozilla/5.0' } }, (res) => {
            resolve({ url, status: res.statusCode });
        });
        req.on('error', (e) => resolve({ url, status: e.message }));
        req.end();
    });
}

async function testPage(pageUrl) {
    console.log(`\n--- Testing ${pageUrl} ---`);
    const html = await new Promise((resolve) => {
        https.get(pageUrl, (res) => {
            let data = '';
            res.on('data', chunk => data += chunk);
            res.on('end', () => resolve(data));
        });
    });

    const baseMatch = html.match(/<base\s+href=["'](.*?)["']/i);
    const baseHref = baseMatch ? baseMatch[1] : '';
    console.log(`Base tag found: ${baseHref}`);

    const resolvedBase = baseHref ? new URL(baseHref, pageUrl).href : pageUrl;
    console.log(`Resolved base URL: ${resolvedBase}`);

    const resources = [];
    
    // Find links, scripts, imgs
    const linkRegex = /<link[^>]+href=["'](.*?)["']/gi;
    let match;
    while ((match = linkRegex.exec(html)) !== null) {
        resources.push(match[1]);
    }
    const scriptRegex = /<script[^>]+src=["'](.*?)["']/gi;
    while ((match = scriptRegex.exec(html)) !== null) {
        resources.push(match[1]);
    }
    const imgRegex = /<img[^>]+src=["'](.*?)["']/gi;
    while ((match = imgRegex.exec(html)) !== null) {
        resources.push(match[1]);
    }

    for (const res of resources) {
        if (res.startsWith('data:')) continue;
        const fullUrl = new URL(res, resolvedBase).href;
        const result = await checkUrl(fullUrl);
        console.log(`[${result.status}] ${fullUrl}`);
    }
}

(async () => {
    await testPage('https://dev.mkavs.com/Pricing');
    await testPage('https://dev.mkavs.com/Branding');
    await testPage('https://dev.mkavs.com/Support');
})();
